packages/frontend-2/pages/projects/[id]/webhooks/index.vue

```diff
@@ -0,0 +1,252 @@
+                class="text-danger"
+              />
+            </div>
+            <span class="text-foreground opacity-50 text-sm truncate">
+              {{ getHistoryStatusInfo(item) }}
+            </span>
+          </div>
+          <span class="text-foreground opacity-50 text-sm truncate">
+            {{ item.url }}
+          </span>
+        </div>
+      </template>
+
+      <template #triggers="{ item }">
+        <div :class="{ 'opacity-60': !item.enabled }">
+          {{
+            item.triggers
+              .map(
+                (event, index, array) =>
+                  `"${event}"${index < array.length - 1 ? ',' : ''}`
+              )
+              .join(' ')
+          }}
+        </div>
+      </template>
+    </LayoutTable>
+
+    <ProjectWebhooksPageDeleteDialog
+      v-model:open="showDeleteWebhookDialog"
+      :webhook="webhookToModify"
+    />
+
+    <ProjectWebhooksPageCreateOrEditDialog
+      v-model:open="showEditWebhookDialog"
+      :webhook="webhookToModify"
+      :stream-id="projectId"
+      @webhook-created="handleWebhookCreated"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref } from 'vue'
+import { useMutation, useQuery } from '@vue/apollo-composable'
+import {
+  PlusIcon,
+  BookOpenIcon,
+  InformationCircleIcon,
+  CheckCircleIcon,
+  XCircleIcon
+} from '@heroicons/vue/20/solid'
+import { TrashIcon, PencilIcon } from '@heroicons/vue/24/outline'
+import { FormSwitch, ToastNotificationType } from '@speckle/ui-components'
+import { projectWebhooksQuery } from '~~/lib/projects/graphql/queries'
+import { updateWebhookMutation } from '~~/lib/projects/graphql/mutations'
+import { useGlobalToast } from '~~/lib/common/composables/toast'
+import { projectRoute, projectWebhooksRoute } from '~~/lib/common/helpers/route'
+import { WebhookItem } from '~~/lib/projects/helpers/types'
+import {
+  convertThrowIntoFetchResult,
+  getCacheId,
+  getFirstErrorMessage
+} from '~~/lib/common/helpers/graphql'
+
+const projectId = computed(() => route.params.id as string)
+const route = useRoute()
+
+const { result: pageResult, refetch: refetchWebhooks } = useQuery(
+  projectWebhooksQuery,
+  () => ({
+    projectId: projectId.value
+  })
+)
+
+definePageMeta({
+  middleware: ['require-valid-project']
+})
+
+const { triggerNotification } = useGlobalToast()
+const { mutate: updateMutation } = useMutation(updateWebhookMutation)
+
+const webhookToModify = ref<WebhookItem | null>(null)
+const showDeleteWebhookDialog = ref(false)
+const showEditWebhookDialog = ref(false)
+
+const projectName = computed(() => pageResult.value?.project?.name || 'Unknown Project')
+const webhooks = computed<WebhookItem[]>(() => {
+  return pageResult.value?.project?.webhooks?.items || []
+})
+
+const getHistoryStatus = (item: WebhookItem) => {
+  const recentHistory = item.history?.items?.[0]
+  if (recentHistory) {
+    switch (recentHistory.status) {
+      case 1:
```

Comment:
1. It's good practice to always add a `default:` case, in case a new value gets introduced that doesn't get handled by any of the existing cases. Since you don't have one, the function can theoretically also just return `undefined`
2. I suggest using enums for these kinds of cases when you need a set of named constants: https://www.typescriptlang.org/docs/handbook/enums.html

The entire function can be simplified to something like this:

```ts
enum HistoryStatus {
  NoEvents = 0,
  Alert = 1,
  Called = 2,
  Error = 3,
}

const getHistoryStatus = (item: WebhookItem) => {
  const recentHistory = item.history?.items?.[0]
  const status = recentHistory?.status as Optional<HistoryStatus> // we were careful to assign the same numeric values for each status in the enum, so it's OK to do this kind of type cast
  return status || HistoryStatus.NoEvents
}
```

Reply:
https://github.com/specklesystems/speckle-server/pull/1792/commits/982dec4428dca938c24b5f8e104b3560de3dc075